framework/math/quadratures/angular/curvilinear_product_quadrature.cc

```diff
@@ -414,4 +289,170 @@ GLCProductQuadrature2DRZ::MakeHarmonicIndices(unsigned int scattering_order)
+
+  //  weights sum to 2
+  const auto integral_weights =
+    std::accumulate(polar_quad.weights.begin(), polar_quad.weights.end(), 0.0);
+  if (std::abs(integral_weights) > 0)
+  {
+    const auto fac = polar_quad_sum_weights / integral_weights;
+    if (std::abs(fac - 1) > eps)
+      for (auto& w : polar_quad.weights)
+        w *= fac;
+  }
+  else
+    throw std::invalid_argument("GLProductQuadrature1DSpherical: "
+                                "Polar quadrature weights sum to zero.");
+
+  // Defined on range [-1;+1]
+  if (std::abs(polar_quad.GetRange().first - polar_quad_span.first) > eps or
+      std::abs(polar_quad.GetRange().second - polar_quad_span.second) > eps)
+    polar_quad.SetRange(polar_quad_span);
+
+  // Abscissae sorted in ascending order
+  auto lt_qp = [](const Vector3& qp0, const Vector3& qp1) { return qp0[0] < qp1[0]; };
+  if (not std::is_sorted(polar_quad.qpoints.begin(), polar_quad.qpoints.end(), lt_qp))
+    throw std::invalid_argument("GLProductQuadrature1DSpherical: "
+                                "Polar quadrature abscissae not in ascending order.");
+
+  // Existence of zero-weight abscissae at the start and at the end of the interval
+  if (std::abs(polar_quad.weights.front()) > eps and
+      std::abs(polar_quad.qpoints.front()[0] - polar_quad_span.first) > eps)
+  {
+    polar_quad.weights.emplace(polar_quad.weights.begin(), 0);
+    polar_quad.qpoints.emplace(polar_quad.qpoints.begin(), polar_quad_span.first);
+  }
+  if (std::abs(polar_quad.weights.back()) > eps and
+      std::abs(polar_quad.qpoints.back()[0] - polar_quad_span.second) > eps)
+  {
+    polar_quad.weights.emplace(polar_quad.weights.end(), 0);
+    polar_quad.qpoints.emplace(polar_quad.qpoints.end(), polar_quad_span.second);
+  }
+
+  // Product quadrature initialization
+  // Compute weights, abscissae $(0, \vartheta_{p})$ and direction vectors
+  // $\omega_{p} := ((1-\mu_{p}^{2})^{1/2}, 0, \mu_{p})$
+  weights.clear();
+  abscissae.clear();
+  omegas.clear();
+  for (size_t p = 0; p < polar_quad.weights.size(); ++p)
+  {
+    const auto pol_wei = polar_quad.weights[p];
+    const auto pol_abs = polar_quad.qpoints[p][0];
+    const auto pol_com = std::sqrt(1 - pol_abs * pol_abs);
+
+    const auto weight = pol_wei;
+    const auto abscissa = QuadraturePointPhiTheta(0, std::acos(pol_abs));
+    const auto omega = Vector3(pol_com, 0, pol_abs);
+
+    weights.emplace_back(weight);
+    abscissae.emplace_back(abscissa);
+    omegas.emplace_back(omega);
+  }
+  weights.shrink_to_fit();
+  abscissae.shrink_to_fit();
+  omegas.shrink_to_fit();
+
+  // Map of direction indices
+  map_directions_.clear();
+  for (size_t p = 0; p < polar_quad.weights.size(); ++p)
+  {
+    std::vector<unsigned int> vec_directions_p;
+    vec_directions_p.emplace_back(p);
+    map_directions_.emplace(p, vec_directions_p);
+  }
+
+  // Curvilinear product quadrature
+  // Compute additional parametrising factors
```

Comment:
Bri'ish english: "parametrising". Maybe you want to Americani**z**e it for consistency?